pkg/virt-controller/services/renderresources.go

```diff
@@ -537,29 +538,48 @@ func initContainerMinimalRequests() k8sv1.ResourceList {
 }
 
 func hotplugContainerResourceRequirementsForVMI(vmi *v1.VirtualMachineInstance) k8sv1.ResourceRequirements {
-	if vmi.IsCPUDedicated() || vmi.WantsToHaveQOSGuaranteed() {
-		return k8sv1.ResourceRequirements{
-			Limits:   hotplugContainerMinimalLimits(),
-			Requests: hotplugContainerMinimalLimits(),
-		}
-	} else {
-		return k8sv1.ResourceRequirements{
-			Limits:   hotplugContainerMinimalLimits(),
-			Requests: hotplugContainerMinimalRequests(),
-		}
+	return k8sv1.ResourceRequirements{
+		Limits:   hotplugContainerLimits(),
+		Requests: calculateHotplugContainerRequests(vmi),
 	}
 }
 
-func hotplugContainerMinimalLimits() k8sv1.ResourceList {
+func hotplugContainerLimits() k8sv1.ResourceList {
 	return k8sv1.ResourceList{
 		k8sv1.ResourceCPU:    resource.MustParse("100m"),
 		k8sv1.ResourceMemory: resource.MustParse("80M"),
 	}
 }
 
-func hotplugContainerMinimalRequests() k8sv1.ResourceList {
+func calculateHotplugContainerRequests(vmi *v1.VirtualMachineInstance) k8sv1.ResourceList {
+	// In order to calculate the request to limit ratio, use the request to limit ratio
+	// of the VMI. If the VMI has a cpu request to limit ratio of 1, then we will use the
+	// same ratio for the hotplug container. Same for memory. These ratios can be independent
+	// of each other. Requests can never be > limits.
+	limits := hotplugContainerLimits()
+	// Using UnscaledBig here or limit returns as a 0 depending on the scale (cpu slice in milli)
+	cpuLimit := limits.Cpu().AsDec().UnscaledBig().Int64()
+	// Don't need to use unscaled here because mem is giving values > 1 byte
+	memLimit, _ := limits.Memory().AsInt64()
+	resources := vmi.Spec.Domain.Resources
+	defaultCpuRequest := resource.MustParse("10m")
+	defaultMemoryRequest := resource.MustParse("2M")
+
 	return k8sv1.ResourceList{
-		k8sv1.ResourceCPU:    resource.MustParse("10m"),
-		k8sv1.ResourceMemory: resource.MustParse("2M"),
+		k8sv1.ResourceCPU:    calculateVMIResourceRatio(resources.Requests.Cpu(), resources.Limits.Cpu(), &defaultCpuRequest, cpuLimit, resource.Milli),
+		k8sv1.ResourceMemory: calculateVMIResourceRatio(resources.Requests.Memory(), resources.Limits.Memory(), &defaultMemoryRequest, memLimit, 0),
+	}
+}
+
+func calculateVMIResourceRatio(req, lim, defaultValue *resource.Quantity, limitValue int64, scale resource.Scale) resource.Quantity {
+	if req == nil || req.IsZero() || lim == nil || lim.IsZero() {
+		return *defaultValue
 	}
+
+	request := req.AsDec().UnscaledBig().Int64()
+	limit := lim.AsDec().UnscaledBig().Int64()
+	ratio := float64(request) / float64(limit)
+	// Observed ratio needs to be <= limit Ratio, ceil ensures the ratio is always <=
+	val := int64(math.Ceil(float64(limitValue) * ratio))
```

Comment:
Not math.Floor? unless I am too tired already

Reply:
My bad, ceil is right, because you want to bring the request closer to limit in order to reduce the ratio